packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,226 @@
+pragma solidity 0.8.15;
+
+// Safe
+import { GnosisSafe as Safe } from "safe-contracts/GnosisSafe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { OwnerManager } from "safe-contracts/base/OwnerManager.sol";
+
+// Interfaces
+import { ILivenessModule2 } from "interfaces/safe/ILivenessModule2.sol";
+
+/// @title LivenessModule2
+/// @notice This module allows for challenge-based ownership transfer to a fallback owner
+///         when the Safe becomes unresponsive. The fallback owner can initiate a challenge,
+///         and if the Safe doesn't respond within the challenge period, ownership transfers
+///         to the fallback owner.
+/// @dev This is a singleton contract. To use it:
+///      1. The Safe must first enable this module using ModuleManager.enableModule()
+///      2. The Safe must then configure the module by calling enableModule() with parameters
+contract LivenessModule2 is ILivenessModule2 {
+    /// @notice Reserved address used as the previous owner to the first owner in a Safe
+    address public constant SENTINEL_OWNER = address(0x1);
+
+    /// @notice Mapping from Safe address to its configuration
+    mapping(address => ModuleConfig) public safeConfigs;
+
+    /// @notice Mapping from Safe address to active challenge start time (0 if no challenge)
+    mapping(address => uint256) public challengeStartTime;
+
+    /// @notice Semantic version.
+    /// @custom:semver 2.0.0
+    string public constant version = "2.0.0";
+
+    /// @notice Configures the module for a Safe that has already enabled it
+    /// @dev MUST only be callable by a Safe that has enabled this module
+    /// @param _config The configuration parameters for the module
+    function configure(ModuleConfig memory _config) external {
+        if (_config.livenessResponsePeriod == 0 || _config.fallbackOwner == address(0)) {
+            revert LivenessModule2_InvalidParameters();
+        }
+
+        // Check that this module is enabled on the calling Safe
+        Safe safe = Safe(payable(msg.sender));
+        if (!safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        // Store the configuration for this safe
+        safeConfigs[msg.sender] = _config;
+
+        // Clear any existing challenge when configuring/re-configuring
+        // If a challenge exists, it MUST be canceled, including emitting the appropriate events
+        if (challengeStartTime[msg.sender] != 0) {
+            delete challengeStartTime[msg.sender];
+            emit ChallengeCancelled(msg.sender);
+        }
+
+        emit ModuleEnabled(msg.sender, _config.livenessResponsePeriod, _config.fallbackOwner);
+    }
+
+    /// @notice Clears the module configuration for a Safe
+    /// @dev MUST only be executable by a Safe that has DISABLED this module first
+    /// @dev MUST erase the existing liveness_response_period and fallback_owner data related to the calling safe
```

Comment:
General nit for this PR, we're not sticking to the 100 character limit for comments effectively. It'd be nice if we could just make forge fmt apply this automatically but for now need to maintain it manually.

Reply:
I will fix it once all the other changes are done